docs/markdown/Reference-manual.md

```diff
@@ -1725,7 +1725,10 @@ the following methods.
   0.53.0)*](Release-notes-for-0.50.0.md#can-override-dependency)
   specifies that whenever `dependency(name, ...)` is used, Meson should not
   look it up on the system but instead return `dep_object`, which may either be
-  the result of `dependency()` or `declare_dependency()`.
+  the result of `dependency()` or `declare_dependency()`. It takes optional
+  `static` and `native` keyword arguments. Doing this in a subproject allows the
+  parent project to retrieve the dependency without having to know the
+  dependency variable name: `dependency(name, fallback : subproject_name)`.
```

Comment:
I like the addition of simplifying the usage. Do we still need a sentence that this is also needed when a project builds libraries with private interdependendencies or when libraries need to be used in the same version?

Reply:
I think that would get too long for that section of the doc. Those cases are already explained from meson.override_dependency() doc, by linking to the release notes that have all the examples.